packages/react-slider/src/components/Slider/useSliderStyles.ts

```diff
@@ -4,297 +4,132 @@ import type { SliderState } from './Slider.types';
         background: theme.colorBrandBackgroundHover,
       },
-      [`& .${trackClassName}`]: {
-        background: theme.colorBrandBackgroundHover,
-      },
+      '--slider-progress-color': theme.colorBrandBackgroundHover,
     },
     ':active': {
-      cursor: 'grabbing',
       [`& .${thumbClassName}`]: {
         background: theme.colorBrandBackgroundPressed,
       },
-      [`& .${trackClassName}`]: {
-        background: theme.colorBrandBackgroundPressed,
-      },
+      '--slider-progress-color': theme.colorBrandBackgroundPressed,
     },
   }),
 
   disabled: theme => ({
-    cursor: 'not-allowed',
-  }),
-
-  focusIndicator: theme =>
-    createFocusOutlineStyle(theme, { selector: 'focus-within', style: { outlineOffset: '6px' } }),
-});
-
-/**
- * Styles for the slider wrapper slot
- */
-export const useSliderWrapper = makeStyles({
-  sliderWrapper: theme => ({
-    position: 'absolute',
-    ...shorthands.overflow('hidden'),
-  }),
-
-  horizontal: theme => ({
-    left: '0px',
-    right: '0px',
-    top: '0px',
-    minHeight: 'var(--slider-thumb-size)',
-  }),
-
-  vertical: theme => ({
-    top: '0px',
-    bottom: '0px',
-    left: '0px',
-    minWidth: 'var(--slider-thumb-size)',
-  }),
+    '--slider-rail-color': theme.colorNeutralBackgroundDisabled,
+    '--slider-progress-color': theme.colorNeutralForegroundDisabled,
+  }),
+
+  focusIndicatorHorizontal: theme =>
+    createFocusOutlineStyle(theme, {
+      selector: 'focus-within',
+      style: { outlineOffset: { top: '6px', bottom: '6px', left: '10px', right: '10px' } },
+    }),
+  focusIndicatorVertical: theme =>
+    createFocusOutlineStyle(theme, {
+      selector: 'focus-within',
+      style: { outlineOffset: { top: '10px', bottom: '10px', left: '6px', right: '6px' } },
+    }),
 });
 
 /**
  * Styles for the rail slot
  */
 export const useRailStyles = makeStyles({
   rail: theme => ({
-    position: 'absolute',
     ...shorthands.borderRadius(theme.borderRadiusXLarge),
-    boxSizing: 'border-box',
     pointerEvents: 'none',
-  }),
-
-  enabled: theme => ({
-    backgroundColor: theme.colorNeutralStrokeAccessible,
+    gridArea: 'slider',
+    position: 'relative',
+    background: `linear-gradient(
+      var(--slider-rail-direction),
+      var(--slider-rail-color) 0%,
+      var(--slider-rail-color) var(--slider-rail-offset),
+      var(--slider-progress-color) var(--slider-rail-offset),
+      var(--slider-progress-color) calc(var(--slider-rail-offset) + var(--slider-rail-progress)),
+      var(--slider-rail-color) calc(var(--slider-rail-offset) + var(--slider-rail-progress))
+    )`,
+    ':before': {
+      content: "''",
+      position: 'absolute',
+      background: `repeating-linear-gradient(
+        var(--slider-rail-direction),
+        #0000 0%,
+        #0000 calc(var(--slider-rail-steps-percent) - 1px),
+        #fff calc(var(--slider-rail-steps-percent) - 1px),
+        #fff var(--slider-rail-steps-percent)
```

Comment:
Should these colors (`#fff`) come from the theme?

Reply:
Yeah, it should be the background color. i'll update